src/rust/catnap/mod.rs

```diff
@@ -181,281 +174,266 @@ impl CatnapLibOS {
-        let coroutine: Pin<Box<Operation>> = Box::pin(async move {
-            if let Err(e) = queue.prepare_accepted() {
-                warn!("accept() qd={:?}: {:?}", qd, &e);
-                return (qd, OperationResult::Failed(e));
-            }
+
+        Ok(self
+            .get_queue(qd)?
+            .accept(move |yielder: Yielder| -> Result<TaskHandle, Fail> {
+                // Asynchronous accept code.
+                let coroutine: Pin<Box<Operation>> = self.accept_coroutine(qd, yielder)?;
+                // Insert async coroutine into the scheduler.
+                let task_id: String = format!("Catnap::accept for qd={:?}", qd);
+                self.insert_coroutine(task_id, coroutine)
+            })?)
+    }
+
+    /// Asynchronous cross-queue code for accepting a connection. This function returns a coroutine that runs
+    /// asynchronously to accept a connection and performs any necessary mult-queue operations at the libOS-level after
+    /// the accept succeeds or fails.
+    fn accept_coroutine(&self, qd: QDesc, yielder: Yielder) -> Result<Pin<Box<Operation>>, Fail> {
+        let qtable: Rc<RefCell<IoQueueTable<CatnapQueue>>> = self.qtable.clone();
+        let queue: CatnapQueue = self.get_queue(qd)?;
+
+        Ok(Box::pin(async move {
             // Wait for the accept operation to complete.
-            match queue.accept(yielder).await {
+            match queue.do_accept(yielder).await {
                 Ok(new_queue) => {
-                    queue.commit();
                     // It is safe to call except here because the new queue is connected and it should be connected to a
                     // remote address.
                     let addr: SocketAddrV4 = new_queue
                         .remote()
                         .expect("An accepted socket must have a remote address");
-                    let new_qd: QDesc = qtable_ptr.borrow_mut().alloc(new_queue);
+                    let new_qd: QDesc = qtable.borrow_mut().alloc(new_queue);
                     (qd, OperationResult::Accept((new_qd, addr)))
                 },
                 Err(e) => {
-                    queue.abort();
                     warn!("accept() listening_qd={:?}: {:?}", qd, &e);
                     // assert definitely no pending ops on new_qd
                     (qd, OperationResult::Failed(e))
                 },
             }
-        });
-        let task_id: String = format!("Catnap::pop for qd={:?}", qd);
-        self.insert_task_into_scheduler(task_id, qd, &yielder_handle, coroutine)
+        }))
     }
 
-    /// Establishes a connection to a remote endpoint.
-    pub fn connect(&mut self, qd: QDesc, remote: SocketAddrV4) -> Result<QToken, Fail> {
+    /// Synchronous code to establish a connection to a remote endpoint. This function schedules the asynchronous
+    /// co-routine and performs any necessary synchronous, multi-queue operations at the libOS-level before beginning
```

Comment:
In some places, `co-routine` is used in place of `coroutine`. It would be good to stick to one style throughout.